### PR TITLE
Push notifications ios

### PR DESCRIPTION
**Pull request by jcesarmobile**
Moved part of the LocalNotifications code to a common class as both Push and Local are handled by UNUserNotificationCenter

### CODE REVIEW

File: ios/Capacitor/Capacitor/CAPUNUserNotificationCenterDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  }
+  /**
+   * Request permissions to send notifications
+   */
+  public func requestPermissions() {
+    print("native request permissions")
+    // Override point for customization after application launch.
+    let center = UNUserNotificationCenter.current()
+    center.requestAuthorization(options:[.badge, .alert, .sound]) { (granted, error) in
+      // Enable or disable features based on authorization.
```
Comment (mlynch) [inline ios/Capacitor/Capacitor/CAPUNUserNotificationCenterDelegate.swift]: Something supposed to be here?

File: ios/Capacitor/Capacitor/Plugins/PushNotifications.swift [hunk 2]
```diff
@@ -1 +1,83 @@
...
+
+  }
+
+  @objc public func didFailToRegisterForRemoteNotificationsWithError(notification: NSNotification){
+    guard let error = notification.object as? Error else {
+      return
+    }
+    notifyListeners("didFailToRegisterForRemoteNotificationsWithError", data:[
+      "error": error.localizedDescription
+      ])
```
Comment (mlynch) [inline ios/Capacitor/Capacitor/Plugins/PushNotifications.swift]: A few of these dangling `])` at the wrong indentation level

**Comment by jcesarmobile** [comment]
Fixed that and removed some print lines I left behind.

pushNotificationReceived is fired when the notification is received and the app is in foreground, or when you open the app by tapping a notification from notification center.

**Comment by mlynch** [comment]
Saweeeet!

**Pull request merged**